api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -292,6 +296,21 @@ type AgentMachinePool struct {
 	Name string `json:"name"`
 }
 
+type DiskEncryption struct {
+	// Enable/disable disk encryption on master nodes, worker nodes, or all nodes.
+	//
+	// +kubebuilder:default=none
+	// +kubebuilder:validation:Enum=none;all;masters;workers
+	EnableOn *string `json:"enableOn,omitempty"`
+
+	// The disk encryption mode to use.
+	// +kubebuilder:validation:Enum=tpmv2;tang
+	Mode *string `json:"mode,omitempty"`
+
+	// JSON-formatted string containing additional information regarding tang's configuration
+	TangServers string `json:"tangServers,omitempty" gorm:"type:text"`
+}
+
```

Comment:
Can't we just reuse `models.DiskEncryption` here?

Reply:
Unfortunately no. The Kube API uses camelCase `tangServers` as opposed to snake_case `tang_servers` as we have in REST API

Reply:
Feels like a weist, we ask about it in the channel but lets not block that PR on it as it is more of a infra change.